### ISSUE TITLE
[weed mount] stat error after rename folder

### ISSUE THREAD

**Issue by paochiang** [chrislusf/seaweedfs#2068]
**Describe the bug**
after weed mount, rename folder, then stat folder got error

**System Setup**
- List the command line to start "weed master", "weed volume", "weed filer", "weed s3", "weed mount".
```
# setup master
weed master -port=9333 -mdir=/tmp/master0 -defaultReplication=000 -ip=localhost

# setup volume
weed volume -port=8080 -dir=/tmp/volume0 -ip=localhost -mserver=localhost:9333

# setup filer
weed filer -port=8888  -ip=localhost -master=localhost:9333

#weed mount
weed mount -filer=localhost:8888 -cacheCapacityMB=0 -cacheDir=/tmp/cache001 -dir=/root/mnt

# mkdir and ls
root@dev-4:~# mkdir /root/mnt/sub1
root@dev-4:~# ls /root/mnt
sub1

# rename and ls and cd folder
root@dev-4:~# /root/testproject/main
os.Rename success!
os.Stat newName error: stat /root/mnt/newsub1: no such file or directory
root@dev-4:~# ls /root/mnt
newsub1  topics
root@dev-4:~# cd /root/mnt/newsub1
-bash: cd: /root/mnt/newsub1: No such file or directory
```
- OS version
Ubuntu 18.04.4 LTS
- output of `weed version`
version 30GB 2.47 0f7b43a linux amd64
- if using filer, show the content of `filer.toml`
default

main.go
```
package main

import (
	"fmt"
	"os"
)

func main() {
	oldName := "/root/mnt/sub1"
	newName := "/root/mnt/newsub1"

	if err := os.Rename(oldName, newName); err != nil {
		fmt.Println("rename error:", err)
		return
	}
	fmt.Println("os.Rename success!")

	newFileInfo, err := os.Stat(newName)
	if err != nil {
		fmt.Println("os.Stat newName error:", err)
		return
	}
	fmt.Printf("os.Stat newName suc:%+v\n", newFileInfo)

}
```
**Expected behavior**
real rename success

**Screenshots**
![image](https://user-images.githubusercontent.com/28077875/118075967-b32dbf80-b3e3-11eb-8eb8-ade3fc5d216e.png)

**Additional context**
This problem appeared since version: `2.40 release`: https://github.com/chrislusf/seaweedfs/releases/tag/2.40

**Comment by chrislusf** [comment]
Thanks for this test case!

Could you help to verify the fix? And whether it helps for #2065 ?

**Comment by chrislusf** [comment]
btw: how did you find out this is the problem?

**Comment by paochiang** [comment]
> btw: how did you find out this is the problem?

In my use case, I used rename in some places, and just found the problem of rename, and then I feel that this may also cause the docker-registry problem, because the push image error `unknown bolb` is usually the target file(or maybe folder) cannot be found, but this requires testing to verify.

**Comment by paochiang** [comment]
> Thanks for this test case!
> 
> Could you help to verify the fix? And whether it helps for #2065 ?

this issue was fixed, but https://github.com/chrislusf/seaweedfs/issues/2065 still have problem
